cni/network/invoker_cns.go

```diff
@@ -135,7 +135,7 @@ func (invoker *CNSIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, erro
 
 		ncgw := net.ParseIP(info.ncGatewayIPAddress)
 		if ncgw == nil {
-			if (invoker.ipamMode != util.V4Overlay) && (invoker.ipamMode != util.DualStackOverlay) {
+			if (invoker.ipamMode != util.V4Overlay) && (invoker.ipamMode != util.DualStackOverlay) && (invoker.ipamMode != util.Overlay) {
```

Comment:
With my recent changes: https://github.com/Azure/azure-container-networking/pull/1989/files (this PR gets rid of IPv6Mode already, so we do not check ipam mode to see whether it's dualstack overlay or not). Instead we just check whether ipv6Result obj is nil or not.

Hence, we can get rid of util.DualStackOverlay flag, and even util.V4Overlay

Reply:
This (according to previous logic) should run when in v4overlay mode

If we switch it to say "if ipv6Result is nil or not", then we change the logic, and it won't run for v4overlay mode

Maybe I'm missing something, but I believe we don't want to change it to be that logic, is that right?

Reply:
Okay, I believe those checks you mentioned (inject ipv6 routes to Linux, or netsh rules to windows), happen elsewhere, in `cni/network/network.go` and `network/network_windows.go` respectively

But here, we don't want to change the logic, I thought

Is that right @ashvindeodhar ?

Reply:
Discussed offline, this is correct for these two if statements, we will add a third overlay option, change it in AKS-RP to just 'overlay', then come back and remove 'v4overlay' and 'dualstackoverlay' options